src/scripts/core/models/grid.ts

```diff
@@ -145,26 +175,29 @@ export class Grid extends Container {
             return (!token.matched);
         }
 
-        this.columns.forEach(column => {
-            const unmatched = column.getAllTokens().filter(isNotMatched);
-            const matched = column.getAllTokens().filter(isMatched);
+        this.columns.forEach(targetColumn => {
+            
+            const newTokenArray = targetColumn.getAllTokens();
+
+            const matched = targetColumn.getAllTokens().filter(isMatched);
             matched.forEach(token => {
                 token.shuffleSkin();
+                token.highLight();
+                newTokenArray.push(token);
             });
-            const newColumn = column.getAllTokens();
-            matched.forEach(token => {
-                newColumn.push(token);
-            });
+
+            const unmatched = targetColumn.getAllTokens().filter(isNotMatched);
             unmatched.forEach(token => {
-                newColumn.push(token);
+                newTokenArray.push(token);
             });
```

Comment:
I'm not sure exactly how this is working but it looks like your columns are constantly getting bigger each time this is called.

Since you create newTokenArray, by getting all token in the column:
`const newTokenArray = targetColumn.getAllTokens();`

Then you appear to be pushing both the matched and unmatched tokens into the array again. Nothing is ever being removed from it so it's essentially doubling in size each time?

Reply:
I should also point out that you have an issue with referencing here.

Javascript passes objects as references and arrays are objects.
What that means is in `targetColumn.getAllToken()` where it just returns `this.tokens` (an array), it's returning a reference to that array not a copy of it.

So in the above code everytime you change newTokenArray you are also changing the underlying array inside the original column. Which is why you are not seeing an error inside your `replaceAllTokens(newTokens: Token[]): void` error check, because you are passing your reference back into it and effectively comparing the same array to itself.